### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by gormux**
Pep8 refactoring
Docstrings
Python2 deprecation
Python3 versions update
Black formatting

### CODE REVIEW

**Comment by gormux** [comment]
#3 Some of these issues should be corrected with this

**Comment by ultrabug** [comment]
@gormux wow didn't expect to get a PR so quick ;) thanks!

you got conflicts in here tho :(

quick tip : make sure you sync your fork before doing new PRs as explained here: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

**Comment by ultrabug** [comment]
also, you might be interested in the pyupgrade tool that will take care of a bunch of things for you: https://github.com/asottile/pyupgrade/

**Comment by gormux** [comment]
OK, the "sync my fork" was the thing I was searching for and not finding in github, thanks for the link.
I also used pyupgrade, with py3-only to avoid breakink compatibility.
The only change other options will made is converting from .format() to fstring, thus breaking compatibility with everythin <3.6, so I don't think it's worth it.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+language: python
+python:
+  - "3.4"
```
Comment (ultrabug) [inline .travis.yml]: This version is EOL, please drop it

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - "3.5"
+  - "3.6"      # current default Python on Travis CI
+  - "3.7"
+  - "3.8"
+  - "3.8-dev"  # 3.8 development branch
```
Comment (ultrabug) [inline .travis.yml]: useless, maybe add 3.9?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - "nightly"  # nightly build
```
Comment (ultrabug) [inline .travis.yml]: useless imho

File: setup.py [hunk 4]
```diff
@@ -28,42 +36,46 @@ def finalize_options(self):
...
-        'Programming Language :: Python :: 3.2',
-        'Programming Language :: Python :: 3.3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
-        'Topic :: Software Development :: Libraries :: Python Modules',
+        "License :: OSI Approved :: BSD License",
+        "Operating System :: POSIX :: Linux",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.2",
```
Comment (ultrabug) [inline setup.py]: drop everything <3.5 plz?

File: setup.py [hunk 5]
```diff
@@ -28,42 +36,46 @@ def finalize_options(self):
...
+        "Programming Language :: Python :: 3.3",
+        "Programming Language :: Python :: 3.4",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
```
Comment (ultrabug) [inline setup.py:62]: if travis 3.9 passes, add it ?

**Review by ultrabug — CHANGES_REQUESTED** [review]
Thanks, some reviews

File: .travis.yml (hunk 1 above)
Comment (gormux) [reply .travis.yml]: done

File: .travis.yml (hunk 2 above)
Comment (gormux) [reply .travis.yml]: 3.9 not in travis, so 3.9-dev it is

File: setup.py (hunk 4 above)
Comment (gormux) [reply setup.py]: ok

File: setup.py (hunk 5 above)
Comment (gormux) [reply setup.py:62]: done

File: .travis.yml (hunk 3 above)
Comment (gormux) [reply .travis.yml]: I left 3.6, 3.7, 3.8 and 3.9-dev

File: setup.py (hunk 4 above)
Comment (gormux) [reply setup.py]: corrected, as you said <3.5 and not <=3.5

**Comment by gormux** [comment]
I also added example code for #6

**Pull request merged**